### PR TITLE
Hit and Run Teleporter - Explorer specific traitor item

### PR DESCRIPTION
**Pull request by PopGamer45**
## What Does This PR Do
Adds the hit and run teleporter which is basically a syndicate version of the abductors teleporter minus the abduction related actions and it having a 2.5 second teleport time instead of the 8 seconds, leaving only the action for teleporting people on the telepad onto the station and it teleports you back in 45 seconds unless you are handcuffed and (buckled or being pulled).
It is purchased as a kit for 60TC which includes a beacon for spawning the teleporter console and beacon for spawning the telepad itself.
It has a 3 minute cooldown starting from when you teleport onto the station not back and the telepad releases a "Suspicious Bluespace Signal" into the sector, the telepad cant be deconstructed while it is on cooldown.

Both the teleporter console and telepad can be deconstructed and give both illegal and alien research levels, since it is a syndicate adaptation of the abductors teleporter IC wise too.
The console gives illegals 3 and alien 2, and the telepad illegals 2 and alien 1. Which can in total bring research to illegals 3 and alien 3.

Iam not entirely sure about the price and not so much about the research levels, iam not very good when it comes to balance, also i should note that it can teleport multiple mobs.

## Why It's Good For The Game
I think its obvious that every job should have atleast one job specific traitor item and this adds a in my opinion a job specific traitor item which is pretty on theme with the job. While it is pretty powerful i think the suspicious signal and 3 minute cooldown makes it almost quaranteed that security is gonna be waiting for you near the teleporter if you want to use it for the second time.

## Images of changes
dreamseeker_VGdcg6bFcC

## Testing
Buy teleporter, setup teleporter, use teleporter, get teleported back with teleporter.
Repeat process but now handcuff yourself and get pulled by a helpful vulp and dont get teleported back.
Try various combinations of depowering/deconstruction both machines and try using the machine.
Read this testing log in the past tense.

## Changelog
:cl:
add: Adds the hit and run teleporter, a syndicate version of the abductors teleporter which teleports you back minus that it cant abduct other people with you.
/:cl:

### CODE REVIEW

**Comment by Coolrune206** [comment]
Cor Blimey they're givin' traitors unassisted Cult recall!

**Comment by HMBGERDO** [comment]
i'd say remove power usage to let explorers place it anywhere they want. It already has a beacon, making it more difficult is kinda.. uncool
you would need a big preparation to make this thing, i like more the idea of preparation not taking half round time

**Comment by PopGamer45** [comment]
gonna remove those 2 black topright "sticks" from the telepad sprite tomorrow because they look off when there is not a adjacent wall like in the abductor ship, saying this just that anyone who was gonna mention that knows that iam already on it

File: code/game/machinery/syndicate_teleporter.dm [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+		return
+	if(!locate(/mob/living) in src.loc)
+		return
+	cooldown = world.time + cooldown_time
+	new/obj/effect/temp_visual/teleport_abductor/syndi(T)
+	sleep(25)
+	flick("syndy-pad", src)
+	if(linked_console)
+		linked_console.remove_eye_control(user)
+	for(var/mob/living/target in src.loc)
```
Comment (Sirryan2002) [inline code/game/machinery/syndicate_teleporter.dm:172]:
```suggestion
	for(var/mob/living/target in loc)
```

**Review by Sirryan2002 — CHANGES_REQUESTED** [review]

File: code/game/machinery/syndicate_teleporter.dm [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+		return
+	for(var/obj/machinery/computer/camera_advanced/hit_run_teleporter/T in range(1, src))
+		if(T.linked_pad)
+			continue
+		linked_console = T
+		T.linked_pad = src
+		T.tele_in_action.target = src
+		return
+
+/obj/machinery/syndi_telepad/proc/Teleport_Out(mob/living/carbon/target)
```
[the remark this answers is not in the document]
Comment (PopGamer45) [reply code/game/machinery/syndicate_teleporter.dm:146]: isnt it already used there? By snake case you mean underscores between words if iam not wrong, right?

File: code/game/machinery/syndicate_teleporter.dm [hunk 3]
```diff
@@ -0,0 +1,187 @@
+/obj/machinery/computer/camera_advanced/hit_run_teleporter
+	name = "hit and run teleporter console"
+	desc = "A syndicate teleporter \"inspired\" by the abductor's teleportation technology."
```
Comment (Sirryan2002) [inline code/game/machinery/syndicate_teleporter.dm:3]:
```suggestion
	desc = "A syndicate teleporter \"inspired\" by alien abductor teleportation technology."
```

File: code/game/machinery/syndicate_teleporter.dm [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+/obj/machinery/computer/camera_advanced/hit_run_teleporter/proc/try_link_pad()
+	if(linked_pad)
+		return
+	for(var/obj/machinery/syndi_telepad/T in range(1, src))
+		if(T.linked_console || !T.anchored)
+			continue
+		linked_pad = T
+		T.linked_console = src
+		tele_in_action.target = T
+		return
```
Comment (Sirryan2002) [inline code/game/machinery/syndicate_teleporter.dm:49]:
```suggestion
```
unnecessary return statement

File: code/game/machinery/syndicate_teleporter.dm [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+			continue
+		linked_console = T
+		T.linked_pad = src
+		T.tele_in_action.target = src
+		return
+
+/obj/machinery/syndi_telepad/proc/teleport_out(mob/living/carbon/target)
+	if(stat & (BROKEN))
+		return
+	if(target.handcuffed && (target.buckled || target.pulledby))
```
Comment (Sirryan2002) [inline code/game/machinery/syndicate_teleporter.dm:149]: maybe some sort of feedback to the target would be helpful. I.e. "You feel a force trying to pull you back to your telepad... etc"

File: code/game/machinery/syndicate_teleporter.dm [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+		return
+	flick("syndi-pad", src)
+	new /obj/effect/temp_visual/dir_setting/ninja/cloak(get_turf(target), target.dir)
+	do_sparks(10, 0, target.loc)
+	target.forceMove(get_turf(src))
+	do_sparks(10, 0, target.loc)
+	about_to_retrieve = FALSE
```
Comment (Sirryan2002) [inline code/game/machinery/syndicate_teleporter.dm:156]: this doesn't get set back to `FALSE` if the above if statement is met

File: code/game/machinery/syndicate_teleporter.dm [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+	if(stat & (BROKEN))
+		return
+	if(cooldown > world.time)
+		var/timeleft = cooldown - world.time
+		to_chat(user, "<span class='notice'>[src] is still charging, wait [round(timeleft/10)] seconds.</span>")
+		return
+	if(!locate(/mob/living) in loc)
+		return
+	cooldown = world.time + cooldown_time
+	new/obj/effect/temp_visual/teleport_abductor/syndi(T)
```
Comment (Sirryan2002) [inline code/game/machinery/syndicate_teleporter.dm:168]:
```suggestion
	new /obj/effect/temp_visual/teleport_abductor/syndi(T)
```

File: code/game/objects/items/devices/radio/beacon.dm [hunk 8]
```diff
@@ -145,6 +145,15 @@
 	origin_tech = "bluespace=2;syndicate=2"
 	bomb = /obj/machinery/grey_autocloner
 
+/obj/item/beacon/syndicate/bomb/syndi_teleporter
```
Comment (Sirryan2002) [inline code/game/objects/items/devices/radio/beacon.dm:148]: is there a reason this isn't just a subtype of `/obj/item/beacon/syndicate` ?

File: icons/obj/abductor.dmi
Comment (Sirryan2002) [inline icons/obj/abductor.dmi:1]: is there a better file you could put these sprites in? Yeah they're repainted abductor sprites but they're used for a completely different category of items

**Review by Sirryan2002 — CHANGES_REQUESTED** [review]

**Comment by PopGamer45** [comment]
I have switched to Linux and i thought i could run Paradise except it not being able to run TGUI on Linux and continue this PR, but i cant run Paradise at all so iam gonna close this PR because i cant do any bug testing

**Pull request closed without merging**